clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -2707,11 +2734,287 @@ public void testCommitOffsetUnknownMemberId() {
+        assertTrue(coordinator.commitOffsetsSync(offsets, time.timer(Long.MAX_VALUE)));
+
+        client.prepareResponse(offsetCommitResponse(topicName, Uuid.randomUuid(), Errors.GROUP_AUTHORIZATION_FAILED));
+        assertTrue(coordinator.commitOffsetsSync(offsets, time.timer(Long.MAX_VALUE)));
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(strings = { "", "test1" })
+    public void testInvalidTopicIdReturnedByBrokerWhenCommittingOffsetAsync(String topicName) {
+        client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.ensureCoordinatorReady(time.timer(Long.MAX_VALUE));
+
+        Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
+        offsets.put(ti1p.topicPartition(), new OffsetAndMetadata(100L, "metadata1"));
+        offsets.put(ti2p.topicPartition(), new OffsetAndMetadata(200L, "metadata2"));
+
+        // Response data which makes the common part of the responses exercised in the use cases below.
+        OffsetCommitResponseData commonData = new OffsetCommitResponseData()
+            .setTopics(Arrays.asList(
+                new OffsetCommitResponseTopic()
+                    .setName(null)
+                    .setTopicId(ti2p.topicId())
+                    .setPartitions(singletonList(new OffsetCommitResponsePartition().setPartitionIndex(0)))
+            ));
+
+        BiConsumer<OffsetCommitResponse, Class<? extends Exception>> asserter = (response, exceptionType) -> {
+            OffsetCommitResponseTopic topic = response.data().topics().get(0);
+            OffsetCommitResponseData data = commonData.duplicate();
+            data.topics().add(topic);
+            client.prepareResponse(new OffsetCommitResponse(data));
+
+            AtomicBoolean callbackInvoked = new AtomicBoolean();
+            coordinator.commitOffsetsAsync(offsets, (__, exception) -> {
+                if (exceptionType == null)
+                    assertNull(exception);
+                else
+                    assertEquals(exceptionType, exception.getClass());
+                callbackInvoked.set(true);
+            });
+
+            coordinator.invokeCompletedOffsetCommitCallbacks();
+            assertTrue(callbackInvoked.get());
+        };
+
+        // The following offset commit responses are valid and the authorization failure results in failing the
+        // offset commit invocation.
+        asserter.accept(
+            offsetCommitResponse(topicName, ti1p.topicId(), Errors.GROUP_AUTHORIZATION_FAILED),
+            GroupAuthorizationException.class);
+
+        // The following offset commit responses defines a topic incorrectly. The coordinator ignores the topic,
+        // and the group authorization failure is therefore not propagated.
+        asserter.accept(
+            offsetCommitResponse(topicName, Uuid.ZERO_UUID, Errors.GROUP_AUTHORIZATION_FAILED),
+            null);
```

Comment:
This case is not correct as well in my opinion. The caller should get an exception in this case.